experimental/packages/opentelemetry-instrumentation-http/src/utils.ts

```diff
@@ -495,3 +495,27 @@ export const getIncomingRequestAttributesOnResponse = (
   }
   return attributes;
 };
+
+export function headerCapture(type: 'request' | 'response', headers: string[]) {
+  const normalizedHeaders = new Map(headers.map(header => [header.toLowerCase(), header.toLowerCase().replace(/-/g, '_')]));
+
+  return (span: Span, getHeader: (key: string) => undefined | string | string[] | number) => {
+    for (const [capturedHeader, normalizedHeader] of normalizedHeaders) {
+      const value = getHeader(capturedHeader);
+      
+      if (value === undefined) {
+        continue;
+      }
+
+      const key = `http.${type}.header.${normalizedHeader}`;
+
+      if (typeof value === 'string') {
```

Comment:
That could be coalesced with "else".

Plz add new line to the end of the file.

Reply:
Added the newline, but the if needs to stay, otherwise the non-array branch's deduced type is `string | number` and converting it to an array produces `(string | number)[]` which can't be set as an attribute.